app/views/text_components/_text_component_row.html.haml

```diff
@@ -3,6 +3,9 @@
   %td= text_component.from_day
   %td= text_component.to_day
   %td
+    - if status_class_mapping = {'draft' => 'default', 'fact_checked' => 'warning', 'published' => 'success'} and status_class_mapping.key? text_component.publication_status
```

Comment:
don't use `and` for conditionals but for control flow: http://www.virtuouscode.com/2010/08/02/using-and-and-or-in-ruby/